Storage/Cursor.swift

```diff
@@ -49,16 +50,21 @@ public class Cursor: SequenceType {
             return self[nextIndex++]
         }
     }
+
+    public func close() {
+        status = .Closed
+        statusMessage = "Closed"
+    }
```

Comment:
With the suggested `enum`, there will be no need to keep two state variables in sync anymore.